classes/checker.php

```diff
@@ -0,0 +1,299 @@
+        $res->message = (string) $e;
+        return $res;
+    }
+
+    /**
+     * Processes the check and maps its result and status to a resultmessage.
+     * @param check $check
+     * @return resultmessage
+     */
+    private static function process_check_and_get_result(check $check): resultmessage {
+        $res = new resultmessage();
+
+        $checkresult = $check->get_result();
+
+        // Map check result to nagios level.
+        $map = [
+            result::OK => resultmessage::LEVEL_OK,
+            result::NA => resultmessage::LEVEL_OK,
+            result::WARNING => resultmessage::LEVEL_WARN,
+            result::CRITICAL => resultmessage::LEVEL_CRITICAL,
+            result::ERROR => resultmessage::LEVEL_CRITICAL,
+            result::UNKNOWN => resultmessage::LEVEL_UNKNOWN,
+        ];
+
+        // Get the level, or default to unknown.
+        $status = $checkresult->get_status();
+        $res->level = isset($map[$status]) ? $map[$status] : resultmessage::LEVEL_UNKNOWN;
+
+        list($title, $message) = self::process_title_and_message($check->get_name(), $checkresult->get_summary(),
+            $checkresult->get_details());
+        $res->title = $title;
+        $res->message = $message;
+
+        return $res;
+    }
+
+    /**
+     * Parses, cleans and sets up the correct output.
+     * @param string $title
+     * @param string $summary
+     * @param string $details
+     * @return array array of [$title, $message]
+     */
+    private static function process_title_and_message(string $title, string $summary, string $details): array {
+        // Strip tags from summary and details.
+        $summary = self::clean_text($summary);
+        $details = self::clean_text($details);
+
+        // Get all the lines of the message.
+        $messagelines = explode("\n", $summary);
+        $messagelines = array_merge($messagelines, explode("\n", $details));
+
+        // Clean each one.
+        $messagelines = array_map(function($line) {
+            return self::clean_text($line);
+        }, $messagelines);
+
+        // Remove empty lines.
+        $messagelines = array_filter($messagelines);
+
+        // Use the first line in the title.
+        $title .= ": " . array_shift($messagelines);
+
+        // Use the rest in the message.
+        $message = implode("\n", $messagelines);
+
+        return [$title, $message];
+    }
+
+    /**
+     * Cleans the text ready for output.
+     * @param string $text
+     * @return string
+     */
+    private static function clean_text(string $text): string {
+        // Convert any line breaks to newlines.
+        $text = str_replace("<br />", "\n", $text);
+        $text = str_replace("<br>", "\n", $text);
+
+        // Strip tags.
+        $text = strip_tags($text);
+
+        // Clean any pipe characters from the $msg. This is because pipe characters
+        // separate Nagios performance data from log data.
+        $text = str_replace("|", "[pipe]", $text);
```

Comment:
lets swap this with a unicode vertical bar char

Reply:
Done